set.mm

```diff
@@ -98434,6 +98434,345 @@ a Cantor normal form (and injectivity, together with coherence
       UM $.
   $}
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Transitive closure under a relationship
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c TrPred $.
+  $( Define the transitive predecessor class as a class. $)
+  ctrpred $a class TrPred ( R , A , X ) $.
+
+  ${
+    $d R a y $.  $d A a y $.  $d X a y $.
+    $( Define the transitive predecessors of a class ` X ` under a relationship
+       ` R ` and a class ` A ` .  This class can be thought of as the
+       "smallest" class containing all elements of ` A ` that are linked to
+       ` X ` by a chain of ` R ` relationships (see ~ trpredtr and
+       ~ trpredmintr ).  Definition based off of Lemma 4.2 of Don Monk's notes
+       for Advanced Set Theory, which can be found at
+       ~ http://euclid.colorado.edu/~~monkd/setth.pdf .  (Contributed by Scott
```

Comment:
~~monkd -> ~monkd
https?
Your definition and link are from 2011, the Monk pdf is a new version from 2019, I did not find Lemma 4.2 in it.

Reply:
Apparently Prof. Monk now takes the approach outlined in Levy! For right now, Theorem 8.4 is a good approximation to the reference I want.

I fixed the other notes in the latest commit.

Reply:
The ~~ is to escape it for the typesetter.